src/babylon.abstractScene.ts

```diff
@@ -155,6 +155,9 @@ module BABYLON {
 
         /**
         * All of the tranform nodes added to this scene
+        * In the context a the Scene, it is not supposed to be modified manually.
+        * Any addition or removal should be done using the addTransformNode and removeTransformNode Scene methods.
+        * Note also that the order of the TransformNode wihin the array is not significant and might change.

```

Comment:
I added a warning in the comment. I'll update the waht's new after https://github.com/BabylonJS/Babylon.js/pull/5300 is merged